test/common/upstream/upstream_impl_test.cc

```diff
@@ -973,7 +988,9 @@ TEST_F(HostSetImplLocalityTest, MissingWeight) {
   HostsPerLocalitySharedPtr hosts_per_locality =
       makeHostsPerLocality({{hosts_[0]}, {hosts_[1]}, {hosts_[2]}});
   LocalityWeightsConstSharedPtr locality_weights{new LocalityWeights{1, 0, 1}};
-  host_set_.updateHosts({}, {}, hosts_per_locality, hosts_per_locality, locality_weights, {}, {});
+  auto hosts = std::make_shared<const HostVector>(hosts_per_locality->get()[0]);
```

Comment:
Maybe just add a helper `makeHostsFromHostsPerLocality` that unions together all the hosts. That way we won't get any surprise test failures later if we change code that relies on the fact that the hosts in the locality hosts vector are also in the global host list.